### PR TITLE
Model changes for fpe case ref

### PR DESCRIPTION
**Pull request by LukeLoze1**
# Motivation and Context
CaseRef now generated with fixed format encryption, need a new model for this, so case-api needs to be updated to match it.

# What has changed
Case Entity changed, some tests altered to continue working

# How to test?
Should work with the case-processor branch: https://github.com/ONSdigital/census-rm-case-processor/tree/pseudorandom_case_ref_for_2021

# Links
https://trello.com/c/e9UNd6NM/354-2021-performance-caseref-pseudorandom-productionisation-13

# Screenshots (if appropriate):

### CODE REVIEW

File: src/main/java/uk/gov/ons/census/caseapisvc/model/entity/Case.java [hunk 1]
```diff
@@ -87,9 +93,7 @@
   @Enumerated(EnumType.STRING)
   private CaseState state;
 
-  @OneToMany(
-      mappedBy = "caze",
-      cascade = {CascadeType.ALL})
+  @OneToMany(mappedBy = "caze")
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/model/entity/Case.java:96]: this is good. glad you've got rid of nonstandard cascade, which was only present to make testing more convenient

File: src/test/resources/docker-compose.yml [hunk 2]
```diff
@@ -38,6 +38,7 @@ services:
       - SPRING_RABBITMQ_PORT=5672
       - QUEUECONFIG_UAC_QID_CREATED_QUEUE=dummy.uac-qid-created
       - JAVA_OPTS=-Xmx128m -Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5121 -Dspring.profiles.active=dev
+      - HMACKEY=12, 32, 24, 32, 13, 32, 5, 32
```
Comment (NickGrantX) [inline src/test/resources/docker-compose.yml]: needs to change to be base64 now?

**Review by AdamHawtin — APPROVED** [review]
👍 Works with the case processor changes locally

File: src/main/java/uk/gov/ons/census/caseapisvc/model/entity/Case.java [hunk 3]
```diff
@@ -12,24 +11,31 @@
...
 @Data
 @Entity
 @Table(name = "cases")
 public class Case {
 
-  @Id private int caseRef;
+  @Id private UUID caseId;
 
-  @Column private UUID caseId;
+  // This bad boy allows us to ge;nerate a pseudorandom unique (non-colliding) caseRef
```
Comment (ryangrundy7) [inline src/main/java/uk/gov/ons/census/caseapisvc/model/entity/Case.java:24]: Should fix this comment

**Review by ryangrundy7 — APPROVED** [review]
Looks good to me 👍 Not gonna block on the comment but it should be fixed

**Review by NickGrantX — APPROVED** [review]
LGTM 👍

**Pull request merged**